i.sentinel_1.forest.mask/i.sentinel_1.forest.mask.py

```diff
@@ -0,0 +1,313 @@
+# % required: yes
+# %end
+
+# %option G_OPT_R_OUTPUT
+# % key: forest_mask
+# % description: Name of output raster map containing the forest mask
+# % required: yes
+# %end
+
+# %option G_OPT_V_OUTPUT
+# % key: forest_mask_vect
+# % description: Name of output vector map containing the forest mask
+# % required: no
+# %end
+
+# %flag
+# % key: v
+# % description: Vectorize forest map
+# %end
+
+
+import atexit
+import grass.script as grass
+from grass_gis_helpers.cleanup import general_cleanup
+
+# set global variables
+ID = grass.tempname(12)
+ORIG_REGION = f"original_region_{ID}"
+rm_rast = []
+rm_vec = []
+rm_groups = []
+
+
+def cleanup():
+    general_cleanup(
+        rm_rasters=rm_rast,
+        rm_vectors=rm_vec,
+        rm_groups=rm_groups,
+        orig_region=ORIG_REGION,
+    )
+
+
+def calc_vv_vh_ratio_and_rvi(vv_db, vh_db):
+    """Transform SAR images from db to linear scale and
+    calculate VV/Vh ratio and Radar Vegetation Index
+    Args:
+        vv_db (str): Name of VV SAR image (db)
+        vh_db (str): Name of VH SAR image (db)
+    Returns:
+        vv_vh_ratio_lin (str): Name of VV/VH ratio (lin) raster map
+        rvi_lin (str): Name of Radar Vegetation Index (lin) raster map
+    """
+    # transform input SAR images to linear scale
+    sar_img_vv_lin = f"sar_img_vv_lin_{ID}"
+    sar_img_vh_lin = f"sar_img_vh_lin_{ID}"
+    grass.run_command(
+        "r.mapcalc",
+        expression=f"{sar_img_vv_lin}=10^({vv_db}/10.0)",
+        quiet=True,
+    )
+    rm_rast.append(sar_img_vv_lin)
+    grass.run_command(
+        "r.mapcalc",
+        expression=f"{sar_img_vh_lin}=10^({vh_db}/10.0)",
+        quiet=True,
+    )
+    rm_rast.append(sar_img_vh_lin)
+
+    # define VV/VH ratio raster map name
+    vv_vh_ratio_lin = f"vv_vh_ratio_lin_{ID}"
+
+    # calculate VV/VH ratio (lin)
+    grass.run_command(
+        "r.mapcalc",
+        expression=f"{vv_vh_ratio_lin}={sar_img_vv_lin}/{sar_img_vh_lin}",
+        quiet=True,
+    )
+    rm_rast.append(vv_vh_ratio_lin)
+
+    # define Radar Vegetation Index raster map name
+    rvi_lin = f"rvi_lin_{ID}"
+
+    # calculate Radar Vegetation Index (lin)
+    grass.run_command(
+        "r.mapcalc",
+        expression=f"{rvi_lin}=(4*{sar_img_vh_lin})/({sar_img_vv_lin}+{sar_img_vh_lin})",
+        quiet=True,
+    )
+    rm_rast.append(rvi_lin)
+
+    return vv_vh_ratio_lin, rvi_lin
+
+
+def create_forest_segments(seg_group, core_forest_mask):
+    """Create segments and extract forest segments
+    Args:
+        seg_group (str): Name of raster map group to use for segmentation
+        core_forest_mask (str): Name of pixel-based core forest mask
+    Returns:
+        forest_seg_vect (str): Name of forest segments vector map
+    """
+    # segmentation based on group
+    segments = f"segements_{ID}"
+    grass.run_command(
+        "i.segment",
+        group=seg_group,
+        output=segments,
+        threshold=0.017,
+        iterations=30,
+        overwrite=True,
```

Comment:
you can also use the `minsize` parameter here to avoid having tiny segments (if that is not desired)